pkg/cmd/experimental/syncgroups/ad/augmented_ldapinterface_test.go

```diff
@@ -0,0 +1,115 @@
+package ad
+
+import (
+	"reflect"
+	"testing"
+
+	"gopkg.in/ldap.v2"
+
+	"github.com/openshift/origin/pkg/auth/ldaputil"
+	"github.com/openshift/origin/pkg/auth/ldaputil/testclient"
+)
+
+const (
+	groupsBaseDN     string = "ou=groups,dc=example,dc=com"
+	groupsBaseFilter string = "objectClass=groupOfNames"
+
+	testGroupDN string = "cn=testGroup," + groupsBaseDN
+)
+
+func newTestAugmentedADLDAPInterface(client ldap.Client) *AugmentedADLDAPInterface {
+	// below are common test implementations of LDAPInterface fields
+	userQuery := ldaputil.LDAPQuery{
+		BaseDN:       usersBaseDN,
+		Scope:        ldaputil.ScopeWholeSubtree,
+		DerefAliases: ldaputil.DerefAliasesAlways,
+		TimeLimit:    0,
+		Filter:       usersBaseFilter,
+	}
+	groupMembershipAttributes := []string{"memberOf"}
+	userNameAttributes := []string{"email"}
+	groupQuery := ldaputil.LDAPQueryOnAttribute{
+		LDAPQuery: ldaputil.LDAPQuery{
+			BaseDN:       groupsBaseDN,
+			Scope:        ldaputil.ScopeWholeSubtree,
+			DerefAliases: ldaputil.DerefAliasesAlways,
+			TimeLimit:    0,
+			Filter:       groupsBaseFilter,
+		},
+		QueryAttribute: "dn",
+	}
+	groupNameAttributes := []string{"cn"}
+
+	return NewAugmentedADLDAPInterface(testclient.NewConfig(client),
+		userQuery,
+		groupMembershipAttributes,
+		userNameAttributes,
+		groupQuery,
+		groupNameAttributes)
+}
+
+// newDefaultTestGroup returns a new LDAP entry with the following characteristics:
+// dn: cn=testGroup,ou=groups,dc=example,dc=com
+// cn: testGroup
+func newDefaultTestGroup() *ldap.Entry {
```

Comment:
This should accept the `testGroupDN` as argument so that I don't have to know a global var to reason about the tests.